src/memoize-one.ts

```diff
@@ -29,5 +36,5 @@ export default function memoizeOne<
     return lastResult;
   }
 
-  return memoized as ResultFn;
+  return memoized as (this: any, ...newArgs: Args) => Result;
```

Comment:
Is this technically not the same type as `ResultFn`?

Reply:
No because `ResultFn` can be a function with extra properties, e.g.

```ts
const fn = () => {}
fn.foo = 'bar';
```

… whereas this type is strictly the function signature only. That reflects the behaviour of `memoizeOne` because the returned function does not retain function properties.

Reply:
Interesting!

Reply:
I'll aim to look at this soon